node/pkg/accountant/watcher.go

```diff
@@ -85,70 +85,76 @@ func (acct *Accountant) handleEvents(ctx context.Context, evts <-chan tmCoreType
+					evt, err := parseEvent[WasmObservation](acct.logger, event, "wasm-Observation", acct.contract)
 					if err != nil {
-						acct.logger.Error("acctwatcher: failed to parse wasm event", zap.Error(err), zap.Stringer("e.Data", reflect.TypeOf(e.Data)), zap.Any("event", event))
+						acct.logger.Error("acctwatcher: failed to parse wasm transfer event", zap.Error(err), zap.Stringer("e.Data", reflect.TypeOf(e.Data)), zap.Any("event", event))
 						continue
 					}
 
 					eventsReceived.Inc()
-					acct.processPendingTransfer(xfer)
+					acct.processPendingTransfer(evt)
+				} else if event.Type == "wasm-ObservationError" {
+					evt, err := parseEvent[WasmObservationError](acct.logger, event, "wasm-ObservationError", acct.contract)
+					if err != nil {
+						acct.logger.Error("acctwatcher: failed to parse wasm observation error event", zap.Error(err), zap.Stringer("e.Data", reflect.TypeOf(e.Data)), zap.Any("event", event))
+						continue
+					}
+
+					errorEventsReceived.Inc()
+					acct.handleTransferError(evt.Key.String(), evt.Error, "acct: transfer error event received")
 				} else {
-					acct.logger.Debug("acctwatcher: ignoring non-transfer event", zap.String("eventType", event.Type))
+					acct.logger.Debug("acctwatcher: ignoring uninteresting event", zap.String("eventType", event.Type))
 				}
 			}
 		}
 	}
 }
 
-// WasmTransfer represents a transfer event from the smart contract.
-type WasmTransfer struct {
-	TxHashBytes      []byte      `json:"tx_hash"`
-	Timestamp        uint32      `json:"timestamp"`
-	Nonce            uint32      `json:"nonce"`
-	EmitterChain     uint16      `json:"emitter_chain"`
-	EmitterAddress   vaa.Address `json:"emitter_address"`
-	Sequence         uint64      `json:"sequence"`
-	ConsistencyLevel uint8       `json:"consistency_level"`
-	Payload          []byte      `json:"payload"`
-}
+type (
+	// WasmObservation represents a transfer event from the smart contract.
+	WasmObservation Observation
 
-// parseWasmTransfer parses transfer events from the smart contract. All other event types are ignored.
-func parseWasmTransfer(logger *zap.Logger, event tmAbci.Event, contractAddress string) (*WasmTransfer, error) {
-	if event.Type != "wasm-Transfer" {
-		return nil, fmt.Errorf("not a WasmTransfer event: %s", event.Type)
+	// WasmObservationError represents an error event from the smart contract.
+	WasmObservationError struct {
+		Key   ObservationKey `json:"key"`
+		Error string         `json:"error"`
 	}
+)
 
+func parseEvent[T any](logger *zap.Logger, event tmAbci.Event, name string, contractAddress string) (*T, error) {
 	attrs := make(map[string]json.RawMessage)
 	for _, attr := range event.Attributes {
 		if string(attr.Key) == "_contract_address" {
 			if string(attr.Value) != contractAddress {
-				return nil, fmt.Errorf("WasmTransfer event from unexpected contract: %s", string(attr.Value))
+				return nil, fmt.Errorf("wasm-Observation event from unexpected contract: %s", string(attr.Value))
```

Comment:
nit: should replace "wasm-Observation" with "%s" and pass in `name` as the format argument.  Some comment for the fmt.Errorf calls below.

Reply:
Will address this in the next PR.